src/approaches/interactive_learning_approach.py

```diff
@@ -303,3 +328,27 @@ def _score_atom_set_frequency(self, atom_set: Set[GroundAtom]) -> float:
                 _, pos_examples = ground_atom_sets
                 count += 1 if atom_set.issubset(pos_examples) else 0
         return 1.0 / count
+
+    def _score_atom_set_entropy(self, atom_set: Set[GroundAtom],
+                                state: State) -> float:
+        """Score an atom set as the sum of the entropies of each atom's
+        predicate."""
+        entropy_sum = 0.0
+        for atom in atom_set:
+            x = state.vec(atom.objects)
+            logits = self._pred_to_ensemble[atom.predicate.name].logits(x)
+            entropy_sum += utils.entropy(np.mean(logits))
+        return entropy_sum
+
+    def _score_atom_set_bald(self, atom_set: Set[GroundAtom],
+                             state: State) -> float:
+        """Score an atom set as the sum of the BALD objectives of each atom's
+        predicate."""
+        objective = 0.0
+        for atom in atom_set:
+            x = state.vec(atom.objects)
+            logits = self._pred_to_ensemble[atom.predicate.name].logits(x)
+            entropy = utils.entropy(np.mean(logits))
+            objective += entropy - np.mean(
```

Comment:
looks good. btw, did you ever figure out what they were talking about w.r.t. computational advantages here?

Reply:
hm so the original objective looks to maximize the decrease in expected posterior entropy (given the new data point x):
![image](https://user-images.githubusercontent.com/26547725/155405015-85adfea5-52b7-4ae1-bc9b-a58238d868d0.png)
but the problems include: posteriors are high dimensional, calculating entropy is intractable, possibly infinite dimensional parameter spaces, expensive posterior updates.
however, that objective can be rewritten into the BALD (Bayesian Active Learning by Disagreement) objective (which one of these days i will verify for myself :P)
![image](https://user-images.githubusercontent.com/26547725/155405345-d1c6aad7-cb0f-4143-adf4-02a8b17744c3.png)
which calculates entropy in low-dim output space and only requires O(1) posterior updates

Reply:
thank you! I'm not sure if I fully understand the details, but the high level makes sense -- let's talk through this in person. it looks good to me for now.